test/correctness/cross_compilation.cpp

```diff
@@ -37,7 +37,7 @@ int main(int argc, char **argv) {
 
         #ifndef _MSC_VER
         std::string object_name = "test_object_" + t + ".o";
```

Comment:
Can we fix this to not form the string ".obj" via concatenation? It is really nice to be able to grep code for "obj" and find all occurrences and in this case it seems trivial to just append ".o" or ".obj" based on the platform. (Yes, I am aware that the problem is not introduced by this PR, but it seems easy to fix here.)